### DISCUSSION TITLE
Error: cannot load last frame from video

### DISCUSSION THREAD

**Discussion by sarahbeth94** [talmolab/sleap#1703] [Help!] [answered]
I've got my SLEAP models trained but when I try to run inference on 'all frames in current video' I keep getting the error message 'cannot load frame 17963 from ..." it's always the last frame of the video. This is happening on all my trimmed videos that I trimmed in with ffmpeg in Python. I don't know why it can't load but also don't care as long as I can run inference on the majority of the frames it would be fine. Can anyone help me??

<img width="881" alt="error" src="https://github.com/talmolab/sleap/assets/161734680/42631849-d4b5-4e40-83b4-8ed79a51109c">

**Comment by sarahbeth94** [comment]
As a follow up, is there a best way to run inference on only a range of frames from a certain video ?

**Reply by Marielamend** [reply]
Hello @sarahbeth94!

> You can run inference on just a couple of frames if you'd like by using Shift + drag: Select a range of frames and then run inference on the selected clip:

<img width="372" alt="Screenshot 2024-03-13 at 3 03 47 PM" src="https://github.com/talmolab/sleap/assets/120222245/420143da-6339-420b-8535-27e6750ad478">

Best,
Mariela

**Comment by Marielamend** [comment, marked as answer]
Hello @sarahbeth94!

1. In regards to your error "Unable to load frame": 
> We have seen a few problems like this occur when the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works.


2. To answer this question: 
> As a follow up, is there a best way to run inference on only a range of frames from a certain video ?

You can run inference on just a couple of frames if you'd like by using **Shift + drag:** Select a range of frames 
and then run inference on the selected clip:
![Screenshot 2024-02-29 at 4 05 51 PM](https://github.com/talmolab/sleap/assets/120222245/0fddfb8d-fa43-42b9-9ad2-eb57205966a3)

Let us know how that goes!
Best, 
Mariela

**Reply by sarahbeth94** [reply]
Thanks so much, @Marielamend! Reencoding worked!

**Comment by mjin1812** [comment]
Hi! I've been experiencing a similar problem as many others. Unfortunately, re-encoding all my videos with ffmpeg did not work, although I have tried multiple times. Like of @sarahbeth94, I'm okay with having the last few frames unprocessed. I have tested running inference on almost all the frames, save the last few frames of the video using the solution provided above. This seems to work very well. As a feature request, is there anyway to allow for a batch option where you can type in a given frame range to apply as a standard inference range across all videos for batch processing?

**Reply by talmo** [reply]
Hi @mjin1812,

We'll be adding something that should mitigate this issue in the next version (see #1711).

I'd be curious to hear more about the batch frame range use case if you don't mind [describing it in a bit more detail as a new idea](https://github.com/talmolab/sleap/discussions/new?category=ideas)?

Cheers,

Talmo
